packages/repository/src/__tests__/integration/repositories/relation.factory.integration.ts

```diff
@@ -413,6 +361,51 @@ describe('HasManyThrough relation', () => {
       ]),
     );
   });
+
+  it('links a target instance to the source instance', async () => {
+    const item = await cartItemRepo.create({description: 'an item'});
+    let targets = await customerCartItemRepo.find();
+    expect(targets).to.deepEqual([]);
```

Comment:
I might misunderstand the test case here...My understanding is, link/unlink will add/remove the **through instance**, so I am confused why the targets are empty here?

E.g. There is 
- a doctor with id=1
- a patient with id=9
- **`patient.link(doctor)`** will create an appointment with doctorId=1 & patientId=9

In this case, both target(patient) and source(doctor) exist, they are not empty.

Reply:
Line 366 only creates a "Patient" instance through "patientRepository. It is not linked to any "Doctor" yet. That's why it couldn't find any "patient" through the appointments. Once it is linked, the patient and the appointment should be found ( L 370-375).

Reply:
Thank you @agnes512 , had a quick chat, and I realized `customerCartItemRepo` is a relation repository instead of model's repository, then all the tests here make perfect sense.